src/defs.h

```diff
@@ -486,21 +476,31 @@ struct SocketRecvData {
     int cur_size;       // maximum number of bytes for the next chunk
 };
 
+// big enough to store sockaddr_in and sockaddr_in6
+struct sockaddr_store_t {
+    union {
+        sa_family_t ss_family;
+        sockaddr_in addr4;
+        sockaddr_in6 addr6;
+    };
+};
+
 /**
  * @struct fds_data
  * @brief Socket related info
  */
 
 typedef struct fds_data {
-    struct sockaddr_in server_addr; /**< server address information */
+    struct sockaddr_store_t server_addr; /**< server address information */
+    socklen_t server_addr_len;      /**< server address length */
     int is_multicast;               /**< if this socket is multicast */
     int sock_type;                  /**< SOCK_STREAM (tcp), SOCK_DGRAM (udp), SOCK_RAW (ip) */
     int next_fd;
     int active_fd_count; /**< number of active connections (by default 1-for UDP; 0-for TCP) */
     int *
     active_fd_list; /**< list of fd related active connections (UDP has the same fd by default) */
-    struct sockaddr_in *memberships_addr; /**< more servers on the same socket information */
-    struct in_addr mc_source_ip_addr;     /**< message source ip for multicast packet filtering */
+    struct sockaddr_store_t *memberships_addr; /**< more servers on the same socket information */
+    IPAddress mc_source_ip_addr;    /**< message source ip for multicast packet filtering */
```

Comment:
This addition causes a compilation warning with 'memset(tmp, 0, sizeof(struct fds_data));' sockperf.cpp:3046,3346
warning: ‘void* memset(void*, int, size_t)’ clearing an object of non-trivial type ‘fds_data’ {aka ‘struct fds_data’}; use assignment or value-initialization instead [-Wclass-memaccess]
Cast to void* of fs_data* should help.

Reply:
Added cast to `void *`. In a future (RM 2904933) this will be replaced with default initializers for all struct members.